Add test for SSM GetParameterHistory API

kakakakakku opened the pull request:
Hi! I regularly use LocalStack, and thank you for such an awesome product😀

<!-- Please refer to the contribution guidelines before raising a PR: https://github.com/localstack/localstack/blob/master/docs/CONTRIBUTING.md -->

<!-- Why am I raising this PR? Add context such as related issues, PRs, or documentation. -->
## Motivation
Recently, while using LocalStack’s SSM Parameter Store, I checked the API coverage and found that some APIs were not marked in the "Internal Test Suite". So I wanted to contribute by adding tests.
- https://docs.localstack.cloud/references/coverage/coverage_ssm/

<!-- What changes does this PR make? How does LocalStack behave differently now? -->
## Changes
In this Pull Request, I’ve added a test for the SSM `GetParameterHistory` API to improve coverage.

<!-- Optional section: How to test these changes? -->
## Testing
The tests passed in my local environment.

```sh
$ make test TEST_PATH=tests/aws/services/ssm
(snip)
tests/aws/services/ssm/test_ssm.py::TestSSM::test_describe_parameters PASSED                                                                                                                                                                                                                                                                                                                   [  7%]
tests/aws/services/ssm/test_ssm.py::TestSSM::test_put_parameters PASSED                                                                                                                                                                                                                                                                                                                        [ 14%]
tests/aws/services/ssm/test_ssm.py::TestSSM::test_hierarchical_parameter[/<param>//b//c] PASSED                                                                                                                                                                                                                                                                                                [ 21%]
tests/aws/services/ssm/test_ssm.py::TestSSM::test_hierarchical_parameter[<param>/b/c] PASSED                                                                                                                                                                                                                                                                                                   [ 28%]
tests/aws/services/ssm/test_ssm.py::TestSSM::test_get_secret_parameter PASSED                                                                                                                                                                                                                                                                                                                  [ 35%]
tests/aws/services/ssm/test_ssm.py::TestSSM::test_get_inexistent_secret PASSED                                                                                                                                                                                                                                                                                                                 [ 42%]
tests/aws/services/ssm/test_ssm.py::TestSSM::test_get_parameters_and_secrets PASSED                                                                                                                                                                                                                                                                                                            [ 50%]
tests/aws/services/ssm/test_ssm.py::TestSSM::test_get_parameters_by_path_and_filter_by_labels PASSED                                                                                                                                                                                                                                                                                           [ 57%]
tests/aws/services/ssm/test_ssm.py::TestSSM::test_get_parameter_by_arn PASSED                                                                                                                                                                                                                                                                                                                  [ 64%]
tests/aws/services/ssm/test_ssm.py::TestSSM::test_get_parameter_history PASSED                                                                                                                                                                                                                                                                                                                 [ 71%]
tests/aws/services/ssm/test_ssm.py::TestSSM::test_get_inexistent_maintenance_window PASSED                                                                                                                                                                                                                                                                                                     [ 78%]
tests/aws/services/ssm/test_ssm.py::TestSSM::test_trigger_event_on_systems_manager_change[standard] PASSED                                                                                                                                                                                                                                                                                     [ 85%]
tests/aws/services/ssm/test_ssm.py::TestSSM::test_trigger_event_on_systems_manager_change[domain] PASSED                                                                                                                                                                                                                                                                                       [ 92%]
tests/aws/services/ssm/test_ssm.py::TestSSM::test_trigger_event_on_systems_manager_change[path] PASSED                                                                                                                                                                                                                                                                                         [100%]
```

<!-- Optional section: What's left to do before it can be merged? -->
<!--
## TODO

What's left to do:

- [ ] ...
- [ ] ...
-->

Could you please review? Thanks a lot 👍

kakakakakku commented on localstack-core/localstack/testing/pytest/fixtures.py:
```diff
@@ -1663,7 +1663,10 @@ def _create_parameter(**kwargs):
+        try:
+            aws_client.ssm.delete_parameter(Name=param)
+        except Exception as e:
+            LOG.debug("error cleaning up parameter %s: %s", param, e)
```
When I used the existing `create_parameter` method, I got the following error. Because the test calling create_parameter three times with the Overwrite=True parameter.

```
ERROR tests/aws/services/ssm/test_ssm.py::TestSSM::test_get_parameter_history - botocore.errorfactory.ParameterNotFound: An error occurred (ParameterNotFound) when calling the DeleteParameter operation: Parameter param-36e2b5a6 not found.
```

Since the parameter only needs to be deleted once, I added exception handling, referencing the existing implementation😀

https://github.com/localstack/localstack/blob/master/localstack-core/localstack/testing/pytest/fixtures.py#L163-L164

kakakakakku commented:
I have read the CLA Document and I hereby sign the CLA

dominikschubert commented on tests/aws/services/ssm/test_ssm.py:
```diff
@@ -159,6 +159,20 @@ def test_get_parameter_by_arn(self, create_parameter, aws_client, snapshot, clea
+        create_parameter(Name=param_name, Value="test3", Type="String", Overwrite=True)
+
+        response = aws_client.ssm.get_parameter_history(Name=param_name)
+        assert len(response["Parameters"]) == 3
+        assert response["Parameters"][0]["Value"] == "test"
+        assert response["Parameters"][0]["Version"] == 1
+        assert response["Parameters"][2]["Value"] == "test3"
+        assert response["Parameters"][2]["Version"] == 3
```
Typically we'd use our snapshot library in these cases, which will  also catch changes to the API if we re-validate the case in the future. You can find more information on this in our [contribution guidelines](https://github.com/localstack/localstack/tree/master/docs/testing/parity-testing).

Please let me know if you'd like to tackle this yourself, otherwise I can also generate this for you! :+1:

kakakakakku replied:
Thanks! I will add snapshot test soon :)

dominikschubert reviewed (changes requested):
Thanks for your contribution @kakakakakku ! 

Sorry for the delay, but I'd love to help you get this PR over the line. Let me know if you have questions regarding the suggest change.

kakakakakku commented:
@dominikschubert 
Thank you for your feedback 😀 I rebased onto the latest `master` branch as some time had passed since this pull request was created. And I added snapshot tests. If there are any other mistakes, I will fix them soon.
